ee/agent/storage/sqlite/keyvalue_store_sqlite.go

```diff
@@ -0,0 +1,213 @@
+package agentsqlite
+
+import (
+	"context"
+	"database/sql"
+	"embed"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/sqlite"
+	"github.com/golang-migrate/migrate/v4/source/iofs"
+	_ "modernc.org/sqlite"
+)
+
+//go:embed migrations/*.sqlite
+var migrations embed.FS
+
+type SqliteStore struct {
+	conn          *sql.DB
+	rootDirectory string
+}
+
+func NewStore(ctx context.Context, rootDirectory string) (*SqliteStore, error) {
+	conn, err := dbConn(ctx, rootDirectory)
+	if err != nil {
+		return nil, fmt.Errorf("opening startup db in %s: %w", rootDirectory, err)
+	}
+
+	s := &SqliteStore{
+		conn:          conn,
+		rootDirectory: rootDirectory,
+	}
+
+	if err := s.migrate(ctx); err != nil {
+		return nil, fmt.Errorf("migrating the database: %w", err)
+	}
+
+	return s, nil
+}
+
+// dbConn returns a connection to the database in the given rootDirectory.
+// It will create a database there if one does not yet exist.
+func dbConn(ctx context.Context, rootDirectory string) (*sql.DB, error) {
+	// Ensure db exists
+	startupDbFilepath := dbLocation(rootDirectory)
+	if _, err := os.Stat(startupDbFilepath); os.IsNotExist(err) {
```

Comment:
wondering if there's a way that we can also check for corruption or at least an empty file here? Just thinking if this is something we want cleaned up on uninstall, we will likely have to touch an empty file here for windows on install (and i'm not sure what happens from `sql.Open` then, might be fine anyway)

Reply:
Added some tests for this -- it looks like it already appropriately handles the case where the file is empty.

In the case where the database file is not valid, we hit the error during migration: `migrating the database: creating migrate instance: file is not a database (26)`. This is a little later than I expected -- I thought the call to `PingContext` below should've done that validation.

For now in the test I'm just requiring an error, but maybe we want to do some mitigation when we notice the db is corrupt, like delete+recreate?

Reply:
okay nice! I was more concerned about the empty file and unsure how likely the corruption scenario is, but I definitely agree if that's something we want to handle now it would be nice to have a delete/recreate mechanism handy

Reply:
I'm not sure how likely the corruption scenario is either, but probably better to handle it now than wait for it to be a problem. It looks like [pragma quick_check](https://www.sqlite.org/pragma.html#pragma_quick_check) is a good option for verifying -- I'll give that a shot.